components/feature/recents/src/main/java/mozilla/components/feature/recents/RecentAppsStorage.kt

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.recents
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import mozilla.components.feature.recents.adapter.RecentAppAdapter
+import mozilla.components.feature.recents.db.RecentAppEntity
+import mozilla.components.feature.recents.db.RecentAppsDatabase
+
+class RecentAppsStorage(context: Context) {
+
+    companion object {
+        @VisibleForTesting
+        internal const val DEFAULT_COUNT = 0.0
+        private const val DECAY_MULTIPLIER = 0.95
+    }
+
+    internal var database: Lazy<RecentAppsDatabase> = lazy { RecentAppsDatabase.get(context) }
+
+    /**
+     * Increment the value stored in the database for the selected app. Then, apply a decay to all
+     * other apps in the database. We do not need to handle overflow as it's not reasonably expected
+     * to reach Double.MAX_VALUE for users
+     */
+    fun updateRecentApp(selectedAppName: String) {
+        val recentAppEntity = database.value.recentAppsDao().getRecentApp(selectedAppName)
+        recentAppEntity?.let {
+            it.score += 1
+            database.value.recentAppsDao().updateRecentApp(it)
+            applyDecayToRecentAppsExceptFor(it.name)
+        }
+    }
```

Comment:
CC @grigoryk here since he has been working on `frecency` related code for top sites. :)